client/web/antrea-ui/src/routes/summary.tsx

```diff
@@ -42,68 +56,249 @@ function getConditionInfo(conditions: Condition[] | undefined, name: string): [s
     return [condition.status, new Date(condition.lastHeartbeatTime).toLocaleString()];
 }
 
-function controllerPropertyValues(controller: ControllerInfo): string[] {
+// Define functions that return both displayed value and sort value
+type PropertyValuePair = {
+    display: string;
+    sortValue: PropertyValue;
+};
+
+function controllerPropertyValues(controller: ControllerInfo): PropertyValuePair[] {
     const [healthy, lastHeartbeat] = getConditionInfo(controller.controllerConditions, 'ControllerHealthy');
+    const connectedAgents = controller.connectedAgentNum ?? 0;
+    
+    return [
+        { display: controller.metadata.name, sortValue: controller.metadata.name },
+        { display: controller?.version ?? 'Unknown', sortValue: controller?.version ?? 'Unknown' },
+        { display: refToString(controller.podRef), sortValue: refToString(controller.podRef) },
+        { display: refToString(controller.nodeRef), sortValue: refToString(controller.nodeRef) },
+        { display: connectedAgents.toString(), sortValue: connectedAgents },
+        { display: healthy, sortValue: healthy },
+        { display: lastHeartbeat, sortValue: new Date(lastHeartbeat) },
+    ];
+}
+
+function featureGatePropertyValues(featureGate: FeatureGate): PropertyValuePair[] {
     return [
-        controller.metadata.name,
-        controller?.version ?? 'Unknown',
-        refToString(controller.podRef),
-        refToString(controller.nodeRef),
-        (controller.connectedAgentNum??0).toString(),
-        healthy,
-        lastHeartbeat,
+        { display: featureGate.name, sortValue: featureGate.name },
+        { display: featureGate.status, sortValue: featureGate.status },
+        { display: featureGate.version, sortValue: featureGate.version }
     ];
 }
 
-function featureGatePropertyValues(featureGate: FeatureGate): string[] {
-    return [featureGate.name, featureGate.status, featureGate.version];
+// Helper function to extract node type and numeric suffix for proper sorting
+function getNodeNameSortValue(name: string): [string, number] {
+    // Match patterns like "k8s-node-worker-123" or "k8s-node-control-plane-1"
+    const match = name.match(/^(.*?)(control-plane|worker)-(\d+)$/);
+    if (match) {
+        // Extract the node type and number
+        const prefix = match[1];
+        const nodeType = match[2]; // 'control-plane' or 'worker'
+        const numericPart = parseInt(match[3], 10);
+        
+        // Return as tuple with nodeType first (for primary sorting) and number second
+        // This ensures control-plane comes before worker when sorting
+        return [nodeType, numericPart];
+    }
+    // Fall back to a default value if pattern doesn't match
+    return ['unknown', 0];
```

Comment:
why even have a specific pattern here? this won't match most clusters
the code should be generic